src/modlunit/units.cpp

```diff
@@ -1194,7 +1194,7 @@ void nrnunit_dynamic_str(char (&buf)[NRN_BUFSIZE], const char* name, char* u1, c
 #if (defined(LegacyFR) && LegacyFR == 1)
     Sprintf(buf, "static double %s = %g;\n", name, unit_mag());
 #else
-    Sprintf(buf, "static double %s = %.12g;\n", name, unit_mag());
+    Sprintf(buf, "static double %s = %a;\n", name, unit_mag());
```

Comment:
we need a test for this line

Reply:
It means compile 2 new versions of nrn. Where should I do that?

Reply:
Good point from Alex. I think that comparing the output file with a reference one should be enough. Not sure how easy this is to do in a CTest though and whether it's better to do this using a bash script

Reply:
These lines are actually tested by https://github.com/neuronsimulator/nrn/blob/master/test/pynrn/test_units.py however this is not caught by the coverage because I guess we don't run it when running `nrnivmodl`?

Reply:
Might be because it doesn't run through pytest.

Reply:
Yes, I think the problem is that we don't run `nrnivmodl` which executes these lines under code coverage I suppose